grc/core/blocks/_build.py

```diff
@@ -108,12 +110,29 @@ def add_param(**data):
             add_param(id='maxoutbuf', name='Max Output Buffer', dtype='int',
                       hide='part', value='0', category=ADVANCED_PARAM_TAB)
 
+    def check_if_attr_overrides_str(option_attributes):
+        for key, value in six.iteritems(option_attributes):
+            if key in dir(str):
+                return key
+
+        return None
+
     base_params_n = {}
     for param_data in params_raw:
         param_id = param_data['id']
         if param_id in params:
             raise Exception('Param id "{}" is not unique'.format(param_id))
 
+        if 'option_attributes' in param_data:
+            if param_data['dtype'] != 'enum':
+                send_warning('{} - option_attributes are for enums only, ignoring'.format(block_id, param_data.pop('option_attributes')))
+            else:
+                invalid_key = check_if_attr_overrides_str(param_data['option_attributes'])
+
+                if invalid_key:
+                    del param_data['option_attributes'][invalid_key]
+                    send_warning('{} - option_attribute "{}" overrides str, ignoring'.format(block_id, invalid_key, param_id))
+
```

Comment:
How about we move both these added sections into a separate function - to keep things neat.

Reply:
Done!